core/domain/state_domain.py

```diff
@@ -2710,24 +2737,15 @@ def update_interaction_answer_groups(self, answer_groups_list):
         old_content_id_list = [
             answer_group.outcome.feedback.content_id for answer_group in (
                 self.interaction.answer_groups)]
+
         # TODO(yanamal): Do additional calculations here to get the
         # parameter changes, if necessary.
-        for answer_group_dict in answer_groups_list:
-            rule_specs_list = answer_group_dict['rule_specs']
-            if not isinstance(rule_specs_list, list):
-                raise Exception(
-                    'Expected answer group rule specs to be a list, '
-                    'received %s' % rule_specs_list)
+        for answer_group in answer_groups_list:
+            rule_specs_list = answer_group.rule_specs
 
-            answer_group = AnswerGroup(
-                Outcome.from_dict(answer_group_dict['outcome']), [],
-                answer_group_dict['training_data'],
-                answer_group_dict['tagged_skill_misconception_id'])
             interaction_answer_groups.append(answer_group)
```

Comment:
Do we still need the `interaction_answer_groups` variable? (not sure!)

Reply:
Looks like it is required to maintain the previous flow of the method. I'm unable to confirm it too.